### PR TITLE
zephyr: csip: set member lock/release command refactor

### PR DESCRIPTION
**Pull request by piotrnarajowski**
- Add address_count and addr array parameters to set_lock and set_release btp commands.
- Code cleanup for readability
- Preconditions adjustment
Fixes FATAL ERROR in CSIP/CL/SP/BV-07-C, CSIP/CL/SP/BV-03-C, CSIP/CL/SP/BV-04-C, CSIP/CL/SPE/BI-01-C
Merge after https://github.com/zephyrproject-rtos/zephyr/pull/71059

### CODE REVIEW

**Comment by piotrnarajowski** [comment]
@KKopyscinski please review

File: doc/btp_csip.txt [hunk 1]
```diff
@@ -47,21 +47,27 @@ Commands and responses:
 	Opcode 0x04 - Set Coordinator Lock
 
 		Controller Index:	<controller id>
-		Command parameters:     Members count (1 octet)
+		Command parameters:     Address list (varies)
```
Comment (KKopyscinski) [inline doc/btp_csip.txt:50]: I'm not sure if sending list with no length is a good idea. Do we assume it's inferred from total command length? I don't think it's a good practice. I'd keep Members count and add the list behind it. Other BTP commands follow that pattern

**Review by KKopyscinski — CHANGES_REQUESTED** [review]

File: doc/btp_csip.txt (hunk 1 above)
Comment (piotrnarajowski) [reply doc/btp_csip.txt:50]: Actually address count is sent along with Address list, just forgot to write it in documentation. 
`addr_cnt = len(addr_list) if addr_list else 0

 data.extend(struct.pack('b', addr_cnt))`

File: doc/btp_csip.txt [hunk 2]
```diff
@@ -47,21 +47,29 @@ Commands and responses:
 	Opcode 0x04 - Set Coordinator Lock
 
 		Controller Index:	<controller id>
-		Command parameters:     Members count (1 octet)
+		Command parameters:     Address count (varies)
```
Comment (KKopyscinski) [inline doc/btp_csip.txt:50]:
```suggestion
		Command parameters:     Address count (1 octet)
```
This is always 1 octet long

File: doc/btp_csip.txt [hunk 3]
```diff
@@ -47,21 +47,29 @@ Commands and responses:
...
+		and address: ([(0, 'DB:F5:72:56:C9:EF'), (0, 'DB:F5:72:56:C9:EF')].
+		When Address list is not provided, lock procedure will be performed
+		on all set members. During operation, the IUT may send event:
 				CSIP Lock event
 
 	Opcode 0x05 - Set Coordinator Lock Release
 
 		Controller Index:	<controller id>
-		Command parameters:     Members count (1 octet)
+		Command parameters:     Address count (varies)
```
Comment (KKopyscinski) [inline doc/btp_csip.txt:64]: ditto

File: autopts/ptsprojects/zephyr/csip.py [hunk 4]
```diff
@@ -70,7 +70,7 @@ def set_pixits(ptses):
     pts2.set_pixit("CSIP", "TSPX_delete_link_key", "FALSE")
     pts2.set_pixit("CSIP", "TSPX_pin_code", "0000")
     pts2.set_pixit("CSIP", "TSPX_use_dynamic_pin", "FALSE")
-    pts2.set_pixit("CSIP", "TSPX_delete_ltk", "FALSE")
+    pts2.set_pixit("CSIP", "TSPX_delete_ltk", "TRUE")
```
Comment (KKopyscinski) [inline autopts/ptsprojects/zephyr/csip.py:73]: Maybe changes in this file should be in separate commit

File: doc/btp_csip.txt (hunk 2 above)
Comment (piotrnarajowski) [reply doc/btp_csip.txt]: Right... fixed

File: doc/btp_csip.txt (hunk 3 above)
Comment (piotrnarajowski) [reply doc/btp_csip.txt]: Fixed

File: autopts/ptsprojects/zephyr/csip.py (hunk 4 above)
Comment (piotrnarajowski) [reply autopts/ptsprojects/zephyr/csip.py:73]: Done

**Review by KKopyscinski — APPROVED** [review]
LGTM

**Review by sjanc — APPROVED** [review]
